WordPress/src/main/java/org/wordpress/android/ui/jetpack/scan/ScanActivity.kt

```diff
@@ -31,11 +31,9 @@ class ScanActivity : AppCompatActivity(), ScrollableViewInitializedListener {
 
     private var binding: ScanActivityBinding? = null
 
-    override fun onNewIntent(intent: Intent?) {
```

Comment:
> This is another case where there is a suppression on UnknownNullness in the framework source 🤷‍♂️ .

Yea, spot-on, I also researched the codebase and saw half of those `intent` being used as nullable, while other half as non-null (Java and Kotlin files included). Them being already used as `@NonNull` made me take the `risky` decision to mark them all as `@NonNull`, making this signature consistent across the whole codebase. 🤞 

> I wonder if it is safer to leave it as @Nullable?

Obviously it is safer to leave it and change everything else as `@Nullable`, but I wonder if it is better to "fail fast" or "fail silently"... I am usually on the "fail fast" side. 🤷 

> Also, strangely, I'm seeing this error:
> ```
> Overriding method should call `super.onNewIntent`
> ```
> but, I'm not sure if it is related with the signature 🤔 

About this warning, I've also noticed it, but this is unrelated to the signature problem as for the life of me I could get rid of it no matter what I tried. PS: I am also founding this to happen occasionally on other such methods as well. 🤔

Reply:
> I wonder if it is better to "fail fast" or "fail silently"... I am usually on the "fail fast" side.

I'm usually on the "fail faster" side, which, to me, means failing at _compile time_. I don't think that assigning `@Nullable` will be "fail silently", since the compiler would fail to even _build_ the binary.

That said, the wide usage (half, as you mentioned) of this as `@NonNull` is a good indication that this is likely safe.. but as I've mentioned before, I think changes that _could_ introduce a crash should be on a separate PR so that we could also have an associated crash monitoring task. Then, the other improvements (that could not possibly introduce a crash) would be unblocked.

Reply:
I agree on having a separate PR @mkevins ! 👍

PS: With "fail silently" here I didn't mean at compile time, or even at runtime. What I meant is that we would have branches of code that are doing nothing, and then, us developers thinking that those branches are active, we would unnecessary maintain and develop on top of those. I am sorry to use "fail silently" and "fail fast" to mean various things, and all at once, I just didn't want to introduce "something else" to avoid making it harder to follow this discussion.

Reply:
Thanks for the explanation. 👍  I agree that if we wrongly annotate something as `@Nullable` and then add (or even keep) guards in place, that those conditional branches might be (and have always been) dead code.